src/bootstrap/builder.rs

```diff
@@ -203,6 +209,10 @@ impl StepDescription {
             for path in paths {
                 let mut attempted_run = false;
                 for (desc, should_run) in v.iter().zip(&should_runs) {
+                    if let Some(suite) = should_run.is_suite_path(path) {
+                        attempted_run = true;
+                        desc.maybe_run(builder, suite);
+                    }
                     if let Some(pathset) = should_run.pathset_for_path(path) {
```

Comment:
Just for sanity's sake, let's make this an `else` perhaps? I think it'll make it easier to reason about when steps are run.